java/src/jmri/jmrix/loconet/messageinterp/LocoNetMessageInterpret.java

```diff
@@ -2323,6 +2349,157 @@ private static String interpretOpcMultiSenseTranspPresence(LocoNetMessage l, Str
+        * <AD_L>. * When <AD_H> is not 0x7D, * Address is a 14 bit
+        * value; AD_H{6:0} represent the upper 7 bits * of the 14 bit
+        * address.
+         *
+        * Information reverse-engineered by Michael Ricahrdson
+        **/
+
+        String locoAddr = convertToMixed(l.getElement(4), l.getElement(3));
+        int indexValue = (l.getElement(1) & 0x1E)/2; //bits 4-1
+        int dynamicValue = l.getElement(2) + (l.getElement(1) & 0x01) * 128;
+
+        String railcomAdString = convertRailComAD(indexValue, dynamicValue);
+
+        return Bundle.getMessage("LN_MSG_OPC_MULTI_SENSE_TRANSP_RAILCOM_REPORT",
+                locoAddr, railcomAdString);
+    }
+
+    private static String interpretOpcMultiSenseLong(LocoNetMessage l, String reporterPrefix) {
+        /***************************************************
+        * Multi Sense Long RailCom App DYN message (Loconet OpCode 0xE0)
+         * The message bytes as assigned as follows:
+         *
+         * <0xE0> <0x09> <MSL_I> <BLK_L> <AD_H> <AD_L> <RCDV_H> <RCDV_L> <CHK>
+         *
+         * <0xEO> OpCode
+         * <Ox09> Message Length
+         * <MSL_I> is encoded as shown below
+         *  bit 7 always 0
+         *  bits 6-5 (00 = Absent, 01 = Present, 10 = Present with AppDyn Message, 11 = unknown)
+         *  bit 4 ? - unverified - currently part of detection block number logic following Multi Sense Standard
+         *  bits 0-3 block high
+         *
+         * <BLK_L> 11 bit number representing the detection block. Lower 7 bits plus 4 high bits from <MSL_I> {3:0}
+         *
+         * <AD_H> is encoded as shown below:
+         *  When <AD_H> = 0x7D, Address is a 7 bit value defined solely by <AD_L>.
+         *  When <AD_H> is not 0x7D, Address is a 14 bit value;
+         *  AD_H{6:0} represent the upper 7 bits * of the 14 bit address.
+         *
+         * <RCDV_H> is encoded as shown below:
+         *  bit 7 always 0
+         *  bit 6 - Loco direction: 0 = East, 1 = West
+         *  bits 5-1 RailCom App:Dyn Index Value (5 bit value, but expected 6 bits per standard)
+         *  bit 0 RailCom Dynamic Value high bit
+         *
+         * <RCDV_L> {6:0} represent the lower 7 bits of the 8 bit RailCom Dynamic Value.  The 8th bit is bit 0 of <RCDV_H>
+         *
+         * <CHK>
+         *
+         * Information reverse-engineered by Michael Ricahrdson
+         */
+
+        if (l.getElement(1) == 0x09){  // Only process 0xE0 0x09 messages
+            // Transponding Event
+            // get system and user names
+
+            int type = l.getElement(2) & LnConstants.OPC_MULTI_SENSE_MSG;
+            //0x00 = absent
+            //0x20 = present
+            //0x40 = present with App Dyn
+            //0x60 = unknown
+
+            if (type == 0x60) { //uknown at this point
+                return Bundle.getMessage("LN_MSG_OPC_MULTI_SENSE_LONG_UNKNOWN_MESSAGE");
+
+            } else {  //Process 0x00, 0x20, and 0x40
```

Comment:
Technically, the `else {` and its corresponding `}` is redundant, since `return` is the last operation in the preceding `if` block.

Reply:
Interesting.  I removed as probably easier to follow the logic without the `else {` and corresponding `}`.